In [1]:
import pandas as pd
import seaborn as sns
import matplotlib.pyplot as plt
import numpy as np

In [2]:
from IPython.display import HTML

<script>
    function findAncestor (el, name) {
        while ((el = el.parentElement) && el.nodeName.toLowerCase() !== name);
        return el;
    }
    function colorAll(el, textColor) {
        el.style.color = textColor;
        Array.from(el.children).forEach((e) => {colorAll(e, textColor);});
    }
    function setBackgroundImage(src, textColor) {
        var section = findAncestor(document.currentScript, "section");
        if (section) {
            section.setAttribute("data-background-image", src);
    if (textColor) colorAll(section, textColor);
        }
    }
</script>

<script>
    setBackgroundImage("images/coverage.jpg", "white");
</script>

In [3]:
HTML(
'''
<style>
h1.head {
    color: darkblue;
    font-weight: bold;
    text-align: center;
    font-size: 50px
}

h2.head{
    color: darkblue;
    font-weight:bold;
    font-size: 40px


}
h3.head {
    color: darkblue;
    font-weight: bold;
    font-size:30px;
    shadow:True;
      height: 40px;

}

p.gen {
    color: darkblue;
    font-weight: regular;
    font-size: 16px;
    text-align: justify
}
p.head {
     color: darkblue;
    font-weight: bold;
    font-size: 20px;
    text-align: justify
}

p.sum {
     color: darkblue;
    font-weight: regular;
    font-size: 16px;
    text-align: justify;
    background-color:'lightgrey'
}

ul {
    color: darkblue;
    font-weight: regular;
    font-size: 16px;
    text-align: justify
}
</style>
'''


)

In [4]:
def data_survey(df):
    dtypes = df.dtypes
    missing_values = df.isnull().sum()
    missing_ration = (df.isnull().sum()/df.shape[0]*100).round(2)
    nuniques = df.nunique()
    uniques = df.apply(lambda x: x.unique())
    result = pd.DataFrame([dtypes, missing_values, missing_ration, nuniques, uniques]).T
    result.columns = ['Dtypes', 'Missing_values', 'Missing_ration', 'Nuniques','Uniques']
    return result

def filter_labels(string_contain_labels):
    list_items = re.split('(\d+)', string_contain_labels)
    list_items[:-1]
    # items_a1[:-1]
    results = {}
    for i in range(len(list_items)-1):
        if list_items[i].isdigit():
    #         float(items_a1[1])
            continue
        else:
            results.update({int(list_items[i+1]): list_items[i][:-2].strip()})
    return results
    
def single_col_count_chart(df, col):
    summary=sociademographic.groupby(col).count()['Target'].sort_values(ascending=False).reset_index()
    summary['Percent'] = (summary.Target/summary.Target.sum()*100).round(2)
    fig2=px.bar(data_frame=summary, y=col, x='Percent', text='Percent',color_discrete_sequence=[px.colors.qualitative.Bold[7]], title=col, labels=dict(Percent='Percentage of Customers', col=col))
    fig2.update_layout(font_family="Times New Roman",
             font_size=14,
                   font_color='darkblue',
             title_font_family="Times New Roman",
             title_font_color='darkblue',
                  title_font_size=25)
    fig2.update_yaxes()
    return fig2
def single_col_pct_chart(df, col):
    summary=df.groupby(col).count()['Target'].sort_values(ascending=False).reset_index()
    summary['Percent'] = (summary.Target/summary.Target.sum()*100).round(2)
    fig2=px.bar(data_frame=summary, x=col, y='Percent', text='Percent',color_discrete_sequence=[px.colors.qualitative.Bold[7]], title=col, labels=dict(Percent='Percentage of Customers', col=col))
    fig2.update_layout(font_family="Times New Roman",
             font_size=14,
                   font_color='darkblue',
             title_font_family="Times New Roman",
             title_font_color='darkblue',
                  title_font_size=25)
#     fig2.update_yaxes()
    fig2.update_layout(height=500, width=700)
#     fig2.show(renderer='colab')
    return fig2
def col_pct_chart_w_target(df, col):
    summary=df.groupby(['Target', col]).agg(count=('Farmer','count')).sort_values(by='count',ascending=False).reset_index()
    summary['Sum'] = summary.groupby(col)['count'].transform('sum')
    summary['Percent'] = (summary['count']/summary['Sum']*100).round(2)
    fig2=px.bar(data_frame=summary, x=col, y='Percent', text='Percent',color='Target', color_discrete_sequence=[px.colors.qualitative.Bold[7], px.colors.qualitative.Bold[9]], title=col, labels=dict(Percent='Percentage of Customers', col=col))
    fig2.update_layout(font_family="Times New Roman",
             font_size=14,
                   font_color='darkblue',
             title_font_family="Times New Roman",
             title_font_color='darkblue',
                  title_font_size=25)
    fig.update_layout(height=500, width=700)
    return fig2


<h1 class=head>CARAVAN INSURANCE POLICY - EXPLAINATION TASK</h1>y

<center><img src='images/caravan/coverage.jpg'></center>

<h2 class=head>1. BUSINESS UNDERSTANDING</h2>

<p class=head>About insurance sector overview:</p>
<p class=gen>Insurance industry is made up of companies that offer risk management in the form of insurance contracts. There are many types of insurance companies that provide some specific types of insurance products (insurance contracts) including 
life insurance, education insurance, health insurance, auto/home/business/boat/caravan insurance. Each of insurance policies cover the loss from risk of each types of accident/bad event occurs. For example, employees' health insurance offerred by many employers to cover their employees' against the costs of on-the-job accidents. Auto insurace prvent car drivers from anxiety status about car accidents in the life time of car usage. If someone buy auto insurance, they protect themself from loss in case of the accident or damage occurs. Life insurance reimburse a large amount of money as your beneficiaries if you die during the policy's term.
Because of the uncertainty theory, the adverse event can occur even with the very low prevelence. For instance, car accident can occur at least once in 20 years of usage with probability a fatal car crash is 1/3972. Or probability of a fire breaking out on a certain night is 0.05%, an American womans lifetime risk of developing colon and rectal cancer is about 4%. In some countries, such as USA/Japanese or Turkey, earthquake or terrorist attack migh occur, so people can request an insurance to be protect in these cases. Regardless of rare in occurrence, the adverse side of life occur in the bad case. To prevent loss from these situation, people voluntarily buy insurance products. The insurance companies take money from the buyer then invest to other profitable products. The buyers who buy insurance products are individuals (who have stable monthly income, who are rich or strong finance, who own home/automobile/boat...), employers.</p>

<p class=head>Effect of insurance on the Economic.</p>
<p class=gen>The insuance industry is a major contributor to New York State's economy. The industry contributed $73.6 billion to the New York gross state product (GSP) in 2021, accounting for 4.2 percent of the GSP. The insurance coupled with banking accounted for 4.73% of Vietnam's total GDP. In Vietnam, according to GlobalData's Global Insurance Database, personal accident and health (PA&H) insurance was the largest general insurance line, accouting for a 32.2% share of the GWP in 2020 and the segment grew by 5.1 % in 2020 during COVID-19 pandemic. 90% of the VNese population buy mandatory public health insurance, 7% of them own private health insurance in the country. Motor insurance and property insurance are the two following largest line that accounting for 30.5% and  24.8% of general insurance GWP in 2020.</p>

<p class=head>Caravan insurance policy</p>
<p class=gen>Caravan is a type of mobile home, inside this people can decorate, design everything like the mini home to live in. The convinience of the product is it has engine and wheels like a vehicle, so that people can simultaneously move and live in the caravan</p>
<p class=gen>Caravan insurance policy is a product of insurance that protect your caravan against accidental damage, theft, storm, fire and flood.</p>
<img src='images/caravan/caravan.png' alt='Image of a Caravan'>

<p class=head>Reference</p>
<ul>
<li>https://math.stackexchange.com/questions/3687938/probability-of-fire</li>
<li>https://www.mayoclinic.org/diseases-conditions/cancer/in-depth/cancer/art-20044092</li>
<li>https://nyia.org/news-more/impact-of-insurance/insurance-economic-impact/</li>
<li>https://www.globaldata.com/media/insurance/general-insurance-industry-vietnam-reach-3-5bn-2026-forecasts-globaldata/</li>
</ul>

<h2 class=head>2. DATA UNDERSTANDING</h2>

<h3 class=head>1. Load Dataset</h3>

In [5]:
train_dataset  = pd.read_csv('data/train_data.csv', index_col=0)

In [6]:
# train_dataset = df.iloc[:train_df.shape[0], :]

In [7]:
train_dataset.shape

(5822, 86)

In [8]:
train_dataset.head()

Customer Subtype  Number of houses  Avg size household Avg age  \
0  Lower class large families                 1                   3   30-40   
1   Mixed small town dwellers                 1                   2   30-40   
2   Mixed small town dwellers                 1                   2   30-40   
3   Modern, complete families                 1                   3   40-50   
4          Large family farms                 1                   4   30-40   

      Customer main type Roman catholic Protestant Other relition No religion  \
0  Family with grown ups             0%   50 - 62%        1 - 10%    24 - 36%   
1  Family with grown ups        1 - 10%   37 - 49%        1 - 10%    37 - 49%   
2  Family with grown ups             0%   37 - 49%       11 - 23%    37 - 49%   
3         Average Family       11 - 23%   24 - 36%       11 - 23%    37 - 49%   
4                Farmers        1 - 10%   37 - 49%        1 - 10%    37 - 49%   

    Married  ... Number of private accident insurance policies  \
0  76 - 88%  ...                                             0   
1  63 - 75%  ...                                             0   
2  24 - 36%  ...                                             0   
3  50 - 62%  ...                                             0   
4  76 - 88%  ...                                             0   

  Number of family accidents insurance policies  \
0                                             0   
1                                             0   
2                                             0   
3                                             0   
4                                             0   

  Number of disability insurance policies Number of fire policies  \
0                                       0                       1   
1                                       0                       1   
2                                       0                       1   
3                                       0                       1   
4                                       0                       1   

  Number of surfboard policies Number of boat policies  \
0                            0                       0   
1                            0                       0   
2                            0                       0   
3                            0                       0   
4                            0                       0   

  Number of bicycle policies Number of property insurance policies  \
0                          0                                     0   
1                          0                                     0   
2                          0                                     0   
3                          0                                     0   
4                          0                                     0   

  Number of social security insurance policies       Target  
0                                            0  noinsurance  
1                                            0  noinsurance  
2                                            0  noinsurance  
3                                            0  noinsurance  
4                                            0  noinsurance  

[5 rows x 86 columns]

In [9]:
list_important_features = ['Customer Subtype', 'Customer main type', 'Protestant',
       'Other relition', 'No religion', 'Married', 'Other relation', 'Singles',
       'Household without children', 'Household with children',
       'High level education', 'Medium level education',
       'Lower level education', 'High status', 'Middle management',
       'Skilled labourers', 'Unskilled labourers', 'Social class A',
       'Social class B1', 'Social class B2', 'Social class C', 'Rented house',
       'Home owners', '1 car', '2 cars', 'No car', 'National Health Service',
       'Private health insurance', 'Income 30.000', 'Income 30-45.000',
       'Income 45-75.000', 'Income 75-122.000', 'Average income',
       'Purchasing power class', 'Contribution private third party insurance',
       'Contribution car policies', 'Contribution fire policies',
       'Contribution boat policies', 'Number of private third party insurance',
       'Number of car policies', 'Number of fire policies']
sociademographic_cols= list(train_dataset.iloc[:,0:43].columns)
print(len(sociademographic_cols))
sociademographic_cols = [i for i in sociademographic_cols if i in list_important_features]
print(len(sociademographic_cols))

product_ownership = train_dataset[train_dataset.iloc[:,43:].columns]
print(len(sociademographic_cols))
product_ownership_cols = [i for i in product_ownership.columns if i in list_important_features]
print(len(product_ownership_cols))

sociademographic_cols = sociademographic_cols + ['Target', 'Farmer']
product_ownership_cols = product_ownership_cols + ['Target', 'Farmer']
# product_ownership_cols

43
34
34
7


In [10]:
sociademographic=train_dataset[sociademographic_cols]
product_ownership=train_dataset[product_ownership_cols]

In [11]:
train_dataset.to_csv('train_data.csv')

In [12]:
import cufflinks as cf
import chart_studio.plotly as py
import seaborn as sns
import plotly.express as px
%matplotlib inline

from plotly.offline import download_plotlyjs, init_notebook_mode, plot, iplot

init_notebook_mode(connected=True)
cf.go_offline()

from plotly import version

<h3 class=head>2. Caravan Insurance Outcome</h1>

In [13]:
fig=px.pie(data_frame=sociademographic.groupby('Target').count()['Customer Subtype'].sort_values(ascending=False).reset_index(), values='Customer Subtype', names='Target', color_discrete_sequence=[px.colors.qualitative.Bold[7], px.colors.qualitative.Bold[9]], title='Fraction of Outcome Among Customer')
fig.update_layout(font_family="Times New Roman",
             font_size=14,
             title_font_family="Times New Roman",
             title_font_color='darkblue', height=500,width=500)
fig.show(renderer = 'colab')
# fig.write_image('images/target2.png')

<p class =gen>Among 5822 customers included in train dataset, only 5.98% (349 customers) got caravan insurance policy. This proportion is significantly imbalanced with those who did not buy the products (94% - 5473 customers).
<hr>
<hr>
<hr>

<p class=gen style='font-size:20px; font-weight:bold'>In Predictive part, through feature importance analysis we select the most 41 attributes that influence on Target. In these below sections, we only consider characteristics of these features following important order:</p>
<img src='images/'>

<h3 class=head>2. Product Ownership</h1>

In [14]:
train_dataset['Contribution car policies'].unique()

array(['f 1000-4999', 'f 0', 'f 500-999', 'f 5000-9999', 'f 10000-19999',
       'f 200-499'], dtype=object)

In [15]:
fig=single_col_pct_chart(product_ownership, 'Contribution car policies')
fig.update_layout(height=500, width=700)
fig.update_xaxes(categoryorder='array', categoryarray= ['f 0', 'f 200-499', 'f 500-999', 'f 1000-4999', 'f 5000-9999', 'f 10000-19999'])
fig.show(renderer='colab')
fig=col_pct_chart_w_target(product_ownership, 'Contribution car policies')
fig.update_layout(height=550, width=700)
fig.update_xaxes(categoryorder='array', categoryarray= ['f 0', 'f 200-499', 'f 500-999', 'f 1000-4999', 'f 5000-9999', 'f 10000-19999'])
fig.show(renderer='colab')

<p class=gen>Majority of customers contributed amount of 0, 500-999 or 1000-4999 dollars in car policies. And only these three groups of customers bought the caravan insurance product: 1000-4999 (11.3%), 0 (2.53%)  and 500-999 (2.28%).</p>

In [16]:
fig=single_col_pct_chart(product_ownership, 'Contribution fire policies')
fig.update_xaxes(categoryorder='array', categoryarray= ['f 0', 'f 1-49', 'f 50-99', 'f 100-199', 'f 200-499', 'f 500-999', 'f 1000-4999', 'f 5000-9999', 'f 10000-19999'])
fig.update_layout(height=500, width=700)
fig.show(renderer='colab')
fig=col_pct_chart_w_target(product_ownership, 'Contribution fire policies')
fig.update_xaxes(categoryorder='array', categoryarray= ['f 0', 'f 1-49', 'f 50-99', 'f 100-199', 'f 200-499', 'f 500-999', 'f 1000-4999', 'f 5000-9999', 'f 10000-19999'])
fig.update_layout(height=550, width=700)
fig.show(renderer='colab')

<p class=gen>More than 45% of customers contributed 0 dollars on fire insurance. Smaller percentage of customers contributed to fire insurance with group of 50-99, 100-199, 200-499 dollars with 9.19%, 15.8%, 21.6% respectively. Of these groups, 200-499 group has the highest proportion of buying caravan products (12.32%), following by 100-199, 500-999 and 0 dollars with percentage of 7.39%, 5.37% and 4.09%. </p>

In [17]:
fig=single_col_pct_chart(product_ownership, 'Contribution private third party insurance')
fig.update_xaxes(categoryorder='array', categoryarray= ['f 0', 'f 1-49', 'f 50-99', 'f 100-199', 'f 200-499', 'f 500-999', 'f 1000-4999', 'f 5000-9999', 'f 10000-19999'])
fig.update_layout(height=500, width=700)
fig.show(renderer='colab')
fig=col_pct_chart_w_target(product_ownership, 'Contribution private third party insurance')
fig.update_xaxes(categoryorder='array', categoryarray= ['f 0', 'f 1-49', 'f 50-99', 'f 100-199', 'f 200-499', 'f 500-999', 'f 1000-4999', 'f 5000-9999', 'f 10000-19999'])
fig.update_layout(height=550, width=700)
fig.show(renderer='colab')

In [18]:
train_dataset[train_dataset['Contribution private third party insurance'] == 'f 50-99'].shape[0]/train_dataset[train_dataset['Contribution private third party insurance'] != 'f 0'].shape[0]

0.9094017094017094

<p class=gen>About 40% of customers have contribution to private third party insurance. Most of private third party insurance contributors, 90.9% customers contribute amount of 50-99 dollars. The more consumers contribute to private third party insurance, the higher percentage of them bought caravan insurance.</p>

In [19]:
fig=single_col_pct_chart(product_ownership, 'Contribution boat policies')
fig.update_xaxes(categoryorder='array', categoryarray= ['f 0', 'f 1-49', 'f 50-99', 'f 100-199', 'f 200-499', 'f 500-999', 'f 1000-4999', 'f 5000-9999', 'f 10000-19999'])
fig.update_layout(height=500, width=700)
fig.show(renderer='colab')
fig=col_pct_chart_w_target(product_ownership, 'Contribution boat policies')
fig.update_xaxes(categoryorder='array', categoryarray= ['f 0', 'f 1-49', 'f 50-99', 'f 100-199', 'f 200-499', 'f 500-999', 'f 1000-4999', 'f 5000-9999', 'f 10000-19999'])
fig.update_layout(height=550, width=700)
fig.show(renderer='colab')

<p class=gen> Less than 0.6% of customers buy boat insurance. To explain the scarcity of consumers, we must consider more factors such as the ratio of boat owners among the population. The boat is only used along the coastal area and by someone who owns a large ship for far and long trips on the sea. Because of scarcity of boat insurance consumers (less than 35 customers), we can not explore the affect of boat insurance consumers on caravan insurance. </p>

In [20]:
train_dataset.columns

Index(['Customer Subtype', 'Number of houses', 'Avg size household', 'Avg age',
       'Customer main type', 'Roman catholic', 'Protestant', 'Other relition',
       'No religion', 'Married', 'Living together', 'Other relation',
       'Singles', 'Household without children', 'Household with children',
       'High level education', 'Medium level education',
       'Lower level education', 'High status', 'Entrepreneur', 'Farmer',
       'Middle management', 'Skilled labourers', 'Unskilled labourers',
       'Social class A', 'Social class B1', 'Social class B2',
       'Social class C', 'Social class D', 'Rented house', 'Home owners',
       '1 car', '2 cars', 'No car', 'National Health Service',
       'Private health insurance', 'Income 30.000', 'Income 30-45.000',
       'Income 45-75.000', 'Income 75-122.000', 'Income>123.000',
       'Average income', 'Purchasing power class',
       'Contribution private third party insurance',
       'Contribution third party insurance (firms)',

<h3 class=head>2. Sociademographic</h3>

<p class=head>Houses and Household</p>

In [21]:
fig=single_col_pct_chart(train_dataset, 'Number of houses')
fig.update_xaxes(categoryorder='array', categoryarray= ['f 0', 'f 1-49', 'f 50-99', 'f 100-199', 'f 200-499', 'f 500-999', 'f 1000-4999', 'f 5000-9999', 'f 10000-19999'])
fig.update_layout(height=500, width=700)
fig.show(renderer='colab')
fig=col_pct_chart_w_target(train_dataset, 'Number of houses')
fig.update_xaxes(categoryorder='array', categoryarray= ['f 0', 'f 1-49', 'f 50-99', 'f 100-199', 'f 200-499', 'f 500-999', 'f 1000-4999', 'f 5000-9999', 'f 10000-19999'])
fig.update_layout(height=550, width=700)
fig.show(renderer='colab')

<p class=gen>More than 99% of customers owned 1-2 houses. 100% caravan buyers owned 1-2 houses.</p>

In [22]:
fig=single_col_pct_chart(train_dataset, 'Avg size household')
fig.update_xaxes(categoryorder='array', categoryarray= ['f 0', 'f 1-49', 'f 50-99', 'f 100-199', 'f 200-499', 'f 500-999', 'f 1000-4999', 'f 5000-9999', 'f 10000-19999'])
fig.update_layout(height=500, width=700)
fig.show(renderer='colab')
fig=col_pct_chart_w_target(train_dataset, 'Avg size household')
fig.update_xaxes(categoryorder='array', categoryarray= ['f 0', 'f 1-49', 'f 50-99', 'f 100-199', 'f 200-499', 'f 500-999', 'f 1000-4999', 'f 5000-9999', 'f 10000-19999'])
fig.update_layout(height=550, width=700)
fig.show(renderer='colab')

<p class=gen>Most of household have 2-4 members (about 95%), The large size of household, the more ability customers buy caravan insurance.</p>

<p class=head>Type of Customer's Family</p>

In [23]:
fig=single_col_pct_chart(sociademographic, 'Customer main type')
fig.update_layout(height=500, width=700)
fig.show(renderer='colab')
fig=col_pct_chart_w_target(sociademographic, 'Customer main type')
fig.update_layout(height=550, width=700)
fig.show(renderer='colab')

<p class=sum>Approximately 53.5% Customers are Family with grown ups/Average or Conservative. Customers whoes families are Living well/Successful hedonists/Driven Growers/Retired and Religous account for from 8 to 9% of customers.</p>
<p class=sum>Among these customers, Driven Growers and Successful hedonists are the 2 groups that bought caravan insurance with the highest rate (13.15% and 8.7%, respectively). Growers invest caravan storage for storing their products.</p>

In [25]:
summary=sociademographic.groupby(['Customer main type', 'Customer Subtype']).agg(count=('Farmer','count')).sort_values(by='count',ascending=False).reset_index()
summary['Sum'] = summary.groupby('Customer main type')['count'].transform('sum')
summary['Percent'] = (summary['count']/summary['Sum']*100).round(2)
fig2=px.bar(data_frame=summary, x='Customer Subtype', y='count', text='count',color='Customer main type',barmode='group', color_discrete_sequence=px.colors.qualitative.Bold, title='Number of Customers viewed in Main Type and Subtype', labels=dict(count='Number of Customers', col='Customer main type'))
fig2.update_layout(font_family="Times New Roman",
             font_size=14,
                   font_color='darkblue',
             title_font_family="Times New Roman",
             title_font_color='darkblue',
                  title_font_size=25)
fig2.update_traces(width=1)
fig2.update_layout(height=600, width=1500)
fig2.show(renderer='colab')
# fig=single_col_pct_chart(sociademographic, 'Customer Subtype')
# fig.update_layout(height=500, width=1000)
# fig.show(renderer='colab')
fig=col_pct_chart_w_target(sociademographic, 'Customer Subtype')
fig.update_layout(height=700, width=1000)
fig.show(renderer='colab')

<p class=gen>In general, main type and subtype of customer related to each other. We can classify subtype based on main type classification.</p>
<ul>
<li><b>Family with grown ups</b> includes subtypes: <i style = 'color:red'><u>Lower class large families</u></i>; <i style = 'color:orange'>Couples with teens, Married with children; Village families</i>; Large family, employed child; Mixed small town dwellers.</li>
<li><b>Conservative families</b> includes subtypes: <i style = 'color:red'>Traditional families; Large religous families</i>.</li>
    <li><b>Driven Growers</b> includes subtypes: <i style = 'color:red'>Middle class families</i>; Carrer and chidcare; Dinki's (double income no kids).</li>
<li><b>Average families</b> includes subtypes: <i style = 'color:orange'>Moderrn, complete families</i>; Young all american family; Stable family; Family starters; Affluent young families.</li>
    <li><b>Living well </b> includes subtypes: <i style = 'color:orange'>Young and rising</i>; Young, low educated; Mixed apartment dwellers; Ethicakky diverse; and Young urban have-nots.</li>
<li><b>Successful hedonists </b> includes subtypes: <i style = 'color:orange'>High status seniors</i>; High income, expensive child; Very Important Provincials; Affluent senior apartments and some Mixed seniors.</li>
<li>Farmers</b> includes subtypes: <i style = 'color:orange'>Mixed rurals</i> and Large family farms </li>
<li><b>Retired and Religious</b> includes subtypes: Some Mixed seniors; <i style = 'color:orange'>Low income catholics</i>; Religious elderly singles; Porchess seniors in the city; Porchless seniors: no front yard.</li>
<li><b>Cruising Seniors</b> includes subtypes: Young seniors in the city; Seniors in apartments; Own home elderly; Residential elderly.</li>
<li><b>Career Loners</b> includes subtypes: Single youth; Student in apartments; Fresh masters in the city; Senior consmopolitants; Suburban youth.</li>
</ul>
<p class=gen>Subtype with red highlight color is the suptypes with more than 300 customers (lower class large families has the highest number of customers - 810). Subtype with orange highlight color is the subtype with more than 200 customers. All other subtypes have less than 150 customers.</p>
<p class=gen> Among these subtypes, Middle class and Affluent young families are the two subtypes have the highest caravan buyers proprotion (>14%)</p>
<p class=gen> High status seniors; High income expensive child; Career and childcare are th three subtype have the percentage of buyers more than 10%.</p>
<p class=gen> Career Loners group (who live and work alone) includes Single youth; Students in apartments; Young urban have-nots; Fresh masters in the city; Senior cosmopholitans and Suburban youth are 6 subtypes did not buy caravan policies. Common feature of these group are young and poor (because of low income or no income).</p>
<p class=gen> All other groups have percentage of 5-8% customers bought caravan insuranceproducts.</p>

<p class=head>Type of Customer's Religion</p>

In [26]:
sociademographic.columns

Index(['Customer Subtype', 'Customer main type', 'Protestant',
       'Other relition', 'No religion', 'Married', 'Other relation', 'Singles',
       'Household without children', 'Household with children',
       'High level education', 'Medium level education',
       'Lower level education', 'High status', 'Middle management',
       'Skilled labourers', 'Unskilled labourers', 'Social class A',
       'Social class B1', 'Social class B2', 'Social class C', 'Rented house',
       'Home owners', '1 car', '2 cars', 'No car', 'National Health Service',
       'Private health insurance', 'Income 30.000', 'Income 30-45.000',
       'Income 45-75.000', 'Income 75-122.000', 'Average income',
       'Purchasing power class', 'Target', 'Farmer'],
      dtype='object')

In [27]:
fig=single_col_pct_chart(sociademographic, 'Protestant')
fig.update_xaxes(categoryorder='array', categoryarray= [ '0%', '1 - 10%', '11 - 23%','24 - 36%','37 - 49%', '50 - 62%','63 - 75%','76 - 88%', '89 - 99%', '100%'])
fig.update_layout(height=500, width=700)
fig.show(renderer='colab')
fig=single_col_pct_chart(sociademographic, 'Other relition')
fig.update_xaxes(categoryorder='array', categoryarray= [ '0%', '1 - 10%', '11 - 23%','24 - 36%','37 - 49%', '50 - 62%','63 - 75%','76 - 88%', '89 - 99%', '100%'])
fig.update_layout(height=500, width=700)
fig.show(renderer='colab')
fig=single_col_pct_chart(sociademographic, 'No religion')
fig.update_xaxes(categoryorder='array', categoryarray= [ '0%', '1 - 10%', '11 - 23%','24 - 36%','37 - 49%', '50 - 62%','63 - 75%','76 - 88%', '89 - 99%', '100%'])
fig.update_layout(height=500, width=700)
fig.show(renderer='colab')
fig=col_pct_chart_w_target(sociademographic, 'Protestant')
fig.update_xaxes(categoryorder='array', categoryarray= [ '0%', '1 - 10%', '11 - 23%','24 - 36%','37 - 49%', '50 - 62%','63 - 75%','76 - 88%', '89 - 99%', '100%'])
fig.update_layout(height=500, width=700)
fig.show(renderer='colab')
fig=col_pct_chart_w_target(sociademographic, 'Other relition')
fig.update_xaxes(categoryorder='array', categoryarray= [ '0%', '1 - 10%', '11 - 23%','24 - 36%','37 - 49%', '50 - 62%','63 - 75%','76 - 88%', '89 - 99%', '100%'])
fig.update_layout(height=500, width=700)
fig.show(renderer='colab')
fig=col_pct_chart_w_target(sociademographic, 'No religion')
fig.update_xaxes(categoryorder='array', categoryarray= [ '0%', '1 - 10%', '11 - 23%','24 - 36%','37 - 49%', '50 - 62%','63 - 75%','76 - 88%', '89 - 99%', '100%'])
fig.update_layout(height=500, width=700)
fig.show(renderer='colab')
# fig=col_pct_chart_w_target(sociademographic, 'Protestant')
# fig.update_layout(height=500, width=700)
# fig.show(renderer='colab')

<p class=gen> Protestant was dominant compared to no religion, few customers followed other religion. People are Protestant or following other religion tend to bought caravan with higher proportion than ones following no religion.</p>

<p class=head>Type of Customer's Marital Status</p>

In [28]:
fig=single_col_pct_chart(sociademographic, 'Married')
fig.update_layout(height=500, width=700)
fig.update_xaxes(categoryorder='array', categoryarray= [ '0%', '1 - 10%', '11 - 23%','24 - 36%','37 - 49%', '50 - 62%','63 - 75%','76 - 88%', '89 - 99%', '100%'])
fig.show(renderer='colab')
fig=single_col_pct_chart(sociademographic, 'Other relation')
fig.update_xaxes(categoryorder='array', categoryarray= [ '0%', '1 - 10%', '11 - 23%','24 - 36%','37 - 49%', '50 - 62%','63 - 75%','76 - 88%', '89 - 99%', '100%'])
fig.update_layout(height=500, width=700)
fig.show(renderer='colab')
fig=single_col_pct_chart(sociademographic, 'Singles')
fig.update_xaxes(categoryorder='array', categoryarray= [ '0%', '1 - 10%', '11 - 23%','24 - 36%','37 - 49%', '50 - 62%','63 - 75%','76 - 88%', '89 - 99%', '100%'])
fig.update_layout(height=500, width=700)
fig.show(renderer='colab')
fig=col_pct_chart_w_target(sociademographic, 'Married')
fig.update_xaxes(categoryorder='array', categoryarray= [ '0%', '1 - 10%', '11 - 23%','24 - 36%','37 - 49%', '50 - 62%','63 - 75%','76 - 88%', '89 - 99%', '100%'])
fig.update_layout(height=500, width=700)
fig.show(renderer='colab')
fig=col_pct_chart_w_target(sociademographic, 'Other relation')
fig.update_xaxes(categoryorder='array', categoryarray= [ '0%', '1 - 10%', '11 - 23%','24 - 36%','37 - 49%', '50 - 62%','63 - 75%','76 - 88%', '89 - 99%', '100%'])
fig.update_layout(height=500, width=700)
fig.show(renderer='colab')
fig=col_pct_chart_w_target(sociademographic, 'Singles')
fig.update_xaxes(categoryorder='array', categoryarray= [ '0%', '1 - 10%', '11 - 23%','24 - 36%','37 - 49%', '50 - 62%','63 - 75%','76 - 88%', '89 - 99%', '100%'])
fig.update_layout(height=500, width=700)
fig.show(renderer='colab')

<p class=gen>Married customers are crucial part of caravan campaign. Those who married tend to buy caravan with higher pecentage than those in other relation or single.</p>

<p class=head>Children and Proportion of Caravan Insurance Purchase</p>

In [29]:
fig=single_col_pct_chart(sociademographic, 'Household without children')
fig.update_xaxes(categoryorder='array', categoryarray= [ '0%', '1 - 10%', '11 - 23%','24 - 36%','37 - 49%', '50 - 62%','63 - 75%','76 - 88%', '89 - 99%', '100%'])
fig.update_layout(height=500, width=700)
fig.show(renderer='colab')
fig=single_col_pct_chart(sociademographic, 'Household with children')
fig.update_xaxes(categoryorder='array', categoryarray= [ '0%', '1 - 10%', '11 - 23%','24 - 36%','37 - 49%', '50 - 62%','63 - 75%','76 - 88%', '89 - 99%', '100%'])
fig.update_layout(height=500, width=700)
fig.show(renderer='colab')
fig=col_pct_chart_w_target(sociademographic, 'Household without children')
fig.update_xaxes(categoryorder='array', categoryarray= [ '0%', '1 - 10%', '11 - 23%','24 - 36%','37 - 49%', '50 - 62%','63 - 75%','76 - 88%', '89 - 99%', '100%'])
fig.update_layout(height=500, width=700)
fig.show(renderer='colab')
fig=col_pct_chart_w_target(sociademographic, 'Household with children')
fig.update_xaxes(categoryorder='array', categoryarray= [ '0%', '1 - 10%', '11 - 23%','24 - 36%','37 - 49%', '50 - 62%','63 - 75%','76 - 88%', '89 - 99%', '100%'])
fig.update_layout(height=500, width=700)
fig.show(renderer='colab')

<p class=gen>Household with children are slightly more than those without children and they have higher rate of caravan insurance buyers.</p>

<p class=head>Type of Customer's Education Level</p>

In [30]:
fig=single_col_pct_chart(sociademographic, 'High level education')
fig.update_xaxes(categoryorder='array', categoryarray= [ '0%', '1 - 10%', '11 - 23%','24 - 36%','37 - 49%', '50 - 62%','63 - 75%','76 - 88%', '89 - 99%', '100%'])
fig.update_layout(height=500, width=700)
fig.show(renderer='colab')
fig=single_col_pct_chart(sociademographic, 'Medium level education')
fig.update_xaxes(categoryorder='array', categoryarray= [ '0%', '1 - 10%', '11 - 23%','24 - 36%','37 - 49%', '50 - 62%','63 - 75%','76 - 88%', '89 - 99%', '100%'])
fig.update_layout(height=500, width=700)
fig.show(renderer='colab')
fig=single_col_pct_chart(sociademographic, 'Lower level education')
fig.update_xaxes(categoryorder='array', categoryarray= [ '0%', '1 - 10%', '11 - 23%','24 - 36%','37 - 49%', '50 - 62%','63 - 75%','76 - 88%', '89 - 99%', '100%'])
fig.update_layout(height=500, width=700)
fig.show(renderer='colab')
fig=col_pct_chart_w_target(sociademographic, 'High level education')
fig.update_xaxes(categoryorder='array', categoryarray= [ '0%', '1 - 10%', '11 - 23%','24 - 36%','37 - 49%', '50 - 62%','63 - 75%','76 - 88%', '89 - 99%', '100%'])
fig.update_layout(height=500, width=700)
fig.show(renderer='colab')
fig=col_pct_chart_w_target(sociademographic, 'Medium level education')
fig.update_xaxes(categoryorder='array', categoryarray= [ '0%', '1 - 10%', '11 - 23%','24 - 36%','37 - 49%', '50 - 62%','63 - 75%','76 - 88%', '89 - 99%', '100%'])
fig.update_layout(height=500, width=700)
fig.show(renderer='colab')
fig=col_pct_chart_w_target(sociademographic, 'Lower level education')
fig.update_xaxes(categoryorder='array', categoryarray= [ '0%', '1 - 10%', '11 - 23%','24 - 36%','37 - 49%', '50 - 62%','63 - 75%','76 - 88%', '89 - 99%', '100%'])
fig.update_layout(height=500, width=700)
fig.show(renderer='colab')

<p class=gen>Customers with lower level education are more than those with medium level education. customers with high level education are fewer than other groups. The higher of education (high level of education) the higher percentage of caravan insurance buyers.</p>

<p class=head>Occupation Level</p>

In [31]:
fig=single_col_pct_chart(sociademographic, 'High status')
fig.update_xaxes(categoryorder='array', categoryarray= [ '0%', '1 - 10%', '11 - 23%','24 - 36%','37 - 49%', '50 - 62%','63 - 75%','76 - 88%', '89 - 99%', '100%'])
fig.update_layout(height=500, width=700)
fig.show(renderer='colab')
fig=single_col_pct_chart(sociademographic, 'Middle management')
fig.update_xaxes(categoryorder='array', categoryarray= [ '0%', '1 - 10%', '11 - 23%','24 - 36%','37 - 49%', '50 - 62%','63 - 75%','76 - 88%', '89 - 99%', '100%'])
fig.update_layout(height=500, width=700)
fig.show(renderer='colab')
fig=single_col_pct_chart(sociademographic, 'Skilled labourers')
fig.update_xaxes(categoryorder='array', categoryarray= [ '0%', '1 - 10%', '11 - 23%','24 - 36%','37 - 49%', '50 - 62%','63 - 75%','76 - 88%', '89 - 99%', '100%'])

fig.update_layout(height=500, width=700)
fig.show(renderer='colab')
fig=single_col_pct_chart(sociademographic, 'Unskilled labourers')
fig.update_xaxes(categoryorder='array', categoryarray= [ '0%', '1 - 10%', '11 - 23%','24 - 36%','37 - 49%', '50 - 62%','63 - 75%','76 - 88%', '89 - 99%', '100%'])

fig.update_layout(height=500, width=700)
fig.show(renderer='colab')
fig=col_pct_chart_w_target(sociademographic, 'High status')
fig.update_xaxes(categoryorder='array', categoryarray= [ '0%', '1 - 10%', '11 - 23%','24 - 36%','37 - 49%', '50 - 62%','63 - 75%','76 - 88%', '89 - 99%', '100%'])
fig.update_layout(height=500, width=700)
fig.show(renderer='colab')
fig=col_pct_chart_w_target(sociademographic, 'Middle management')
fig.update_xaxes(categoryorder='array', categoryarray= [ '0%', '1 - 10%', '11 - 23%','24 - 36%','37 - 49%', '50 - 62%','63 - 75%','76 - 88%', '89 - 99%', '100%'])
fig.update_layout(height=500, width=700)
fig.show(renderer='colab')
fig=col_pct_chart_w_target(sociademographic, 'Skilled labourers')
fig.update_xaxes(categoryorder='array', categoryarray= [ '0%', '1 - 10%', '11 - 23%','24 - 36%','37 - 49%', '50 - 62%','63 - 75%','76 - 88%', '89 - 99%', '100%'])
fig.update_layout(height=500, width=700)
fig.show(renderer='colab')
fig=col_pct_chart_w_target(sociademographic, 'Unskilled labourers')
fig.update_xaxes(categoryorder='array', categoryarray= [ '0%', '1 - 10%', '11 - 23%','24 - 36%','37 - 49%', '50 - 62%','63 - 75%','76 - 88%', '89 - 99%', '100%'])
fig.update_layout(height=500, width=700)
fig.show(renderer='colab')

<p class=gen>People with higher status or skilled in occupation tend to buy caravan insurance policies with higher proportion.</p>

<p class=head>Type of Social Class</p>

In [32]:
fig=single_col_pct_chart(sociademographic, 'Social class A')
fig.update_xaxes(categoryorder='array', categoryarray= [ '0%', '1 - 10%', '11 - 23%','24 - 36%','37 - 49%', '50 - 62%','63 - 75%','76 - 88%', '89 - 99%', '100%'])
fig.update_layout(height=500, width=700)
fig.show(renderer='colab')
fig=single_col_pct_chart(sociademographic, 'Social class B1')
fig.update_xaxes(categoryorder='array', categoryarray= [ '0%', '1 - 10%', '11 - 23%','24 - 36%','37 - 49%', '50 - 62%','63 - 75%','76 - 88%', '89 - 99%', '100%'])
fig.update_layout(height=500, width=700)
fig.show(renderer='colab')
fig=single_col_pct_chart(sociademographic, 'Social class B2')
fig.update_xaxes(categoryorder='array', categoryarray= [ '0%', '1 - 10%', '11 - 23%','24 - 36%','37 - 49%', '50 - 62%','63 - 75%','76 - 88%', '89 - 99%', '100%'])
fig.update_layout(height=500, width=700)
fig.show(renderer='colab')
fig=single_col_pct_chart(sociademographic, 'Social class C')
fig.update_xaxes(categoryorder='array', categoryarray= [ '0%', '1 - 10%', '11 - 23%','24 - 36%','37 - 49%', '50 - 62%','63 - 75%','76 - 88%', '89 - 99%', '100%'])
fig.update_layout(height=500, width=700)
fig.show(renderer='colab')
fig=single_col_pct_chart(train_dataset, 'Social class D')
fig.update_xaxes(categoryorder='array', categoryarray= [ '0%', '1 - 10%', '11 - 23%','24 - 36%','37 - 49%', '50 - 62%','63 - 75%','76 - 88%', '89 - 99%', '100%'])
fig.update_layout(height=500, width=700)
fig.show(renderer='colab')
fig=col_pct_chart_w_target(sociademographic, 'Social class A')
fig.update_xaxes(categoryorder='array', categoryarray= [ '0%', '1 - 10%', '11 - 23%','24 - 36%','37 - 49%', '50 - 62%','63 - 75%','76 - 88%', '89 - 99%', '100%'])
fig.update_layout(height=500, width=700)
fig.show(renderer='colab')
fig=col_pct_chart_w_target(sociademographic, 'Social class B1')
fig.update_xaxes(categoryorder='array', categoryarray= [ '0%', '1 - 10%', '11 - 23%','24 - 36%','37 - 49%', '50 - 62%','63 - 75%','76 - 88%', '89 - 99%', '100%'])
fig.update_layout(height=500, width=700)
fig.show(renderer='colab')
fig=col_pct_chart_w_target(sociademographic, 'Social class B2')
fig.update_xaxes(categoryorder='array', categoryarray= [ '0%', '1 - 10%', '11 - 23%','24 - 36%','37 - 49%', '50 - 62%','63 - 75%','76 - 88%', '89 - 99%', '100%'])
fig.update_layout(height=500, width=700)
fig.show(renderer='colab')
fig=col_pct_chart_w_target(sociademographic, 'Social class C')
fig.update_xaxes(categoryorder='array', categoryarray= [ '0%', '1 - 10%', '11 - 23%','24 - 36%','37 - 49%', '50 - 62%','63 - 75%','76 - 88%', '89 - 99%', '100%'])
fig.update_layout(height=500, width=700)
fig.show(renderer='colab')
fig=col_pct_chart_w_target(train_dataset, 'Social class D')
fig.update_xaxes(categoryorder='array', categoryarray= [ '0%', '1 - 10%', '11 - 23%','24 - 36%','37 - 49%', '50 - 62%','63 - 75%','76 - 88%', '89 - 99%', '100%'])
fig.update_layout(height=500, width=700)
fig.show(renderer='colab')

<p class=gen>Number of customers in class C and B are larger gen those in class A and D. Those in Class A tend to prefer buy caravan.</p>

<p class=head>Acommodation Status</p>

In [33]:
fig=single_col_pct_chart(sociademographic, 'Rented house')
fig.update_xaxes(categoryorder='array', categoryarray= [ '0%', '1 - 10%', '11 - 23%','24 - 36%','37 - 49%', '50 - 62%','63 - 75%','76 - 88%', '89 - 99%', '100%'])
fig.update_layout(height=500, width=700)
fig.show(renderer='colab')
fig=single_col_pct_chart(sociademographic, 'Home owners')
fig.update_xaxes(categoryorder='array', categoryarray= [ '0%', '1 - 10%', '11 - 23%','24 - 36%','37 - 49%', '50 - 62%','63 - 75%','76 - 88%', '89 - 99%', '100%'])
fig.update_layout(height=500, width=700)
fig.show(renderer='colab')
fig=col_pct_chart_w_target(sociademographic, 'Rented house')
fig.update_xaxes(categoryorder='array', categoryarray= [ '0%', '1 - 10%', '11 - 23%','24 - 36%','37 - 49%', '50 - 62%','63 - 75%','76 - 88%', '89 - 99%', '100%'])
fig.update_layout(height=500, width=700)
fig.show(renderer='colab')
fig=col_pct_chart_w_target(sociademographic, 'Home owners')
fig.update_xaxes(categoryorder='array', categoryarray= [ '0%', '1 - 10%', '11 - 23%','24 - 36%','37 - 49%', '50 - 62%','63 - 75%','76 - 88%', '89 - 99%', '100%'])
fig.update_layout(height=500, width=700)
fig.show(renderer='colab')

<p class=gen>Customers who owned home are dominant compared to who rent house. Ability to buy caravan insurance of customers who owned home is higher than those who rent house.</p>

<p class=head>Number of Cars Owned</p>

In [34]:
fig=single_col_pct_chart(sociademographic, '1 car')
fig.update_xaxes(categoryorder='array', categoryarray= [ '0%', '1 - 10%', '11 - 23%','24 - 36%','37 - 49%', '50 - 62%','63 - 75%','76 - 88%', '89 - 99%', '100%'])
fig.update_layout(height=500, width=700)
fig.show(renderer='colab')
fig=single_col_pct_chart(sociademographic, '2 cars')
fig.update_xaxes(categoryorder='array', categoryarray= [ '0%', '1 - 10%', '11 - 23%','24 - 36%','37 - 49%', '50 - 62%','63 - 75%','76 - 88%', '89 - 99%', '100%'])
fig.update_layout(height=500, width=700)
fig.show(renderer='colab')
fig=single_col_pct_chart(sociademographic, 'No car')
fig.update_xaxes(categoryorder='array', categoryarray= [ '0%', '1 - 10%', '11 - 23%','24 - 36%','37 - 49%', '50 - 62%','63 - 75%','76 - 88%', '89 - 99%', '100%'])
fig.update_layout(height=500, width=700)
fig.show(renderer='colab')
fig=col_pct_chart_w_target(sociademographic, '1 car')
fig.update_xaxes(categoryorder='array', categoryarray= [ '0%', '1 - 10%', '11 - 23%','24 - 36%','37 - 49%', '50 - 62%','63 - 75%','76 - 88%', '89 - 99%', '100%'])
fig.update_layout(height=500, width=700)
fig.show(renderer='colab')
fig=col_pct_chart_w_target(sociademographic, '2 cars')
fig.update_xaxes(categoryorder='array', categoryarray= [ '0%', '1 - 10%', '11 - 23%','24 - 36%','37 - 49%', '50 - 62%','63 - 75%','76 - 88%', '89 - 99%', '100%'])
fig.update_layout(height=500, width=700)
fig.show(renderer='colab')
fig=col_pct_chart_w_target(sociademographic, 'No car')
fig.update_xaxes(categoryorder='array', categoryarray= [ '0%', '1 - 10%', '11 - 23%','24 - 36%','37 - 49%', '50 - 62%','63 - 75%','76 - 88%', '89 - 99%', '100%'])
fig.update_layout(height=500, width=700)
fig.show(renderer='colab')

<p class=gen> Number of Customers who owned 1 car is higher than those who owned 2 cars or no car. Customers who owned at least a car have higher ability to buy caravan insurance compared to those who are not.</p>

In [35]:
train_dataset.columns

Index(['Customer Subtype', 'Number of houses', 'Avg size household', 'Avg age',
       'Customer main type', 'Roman catholic', 'Protestant', 'Other relition',
       'No religion', 'Married', 'Living together', 'Other relation',
       'Singles', 'Household without children', 'Household with children',
       'High level education', 'Medium level education',
       'Lower level education', 'High status', 'Entrepreneur', 'Farmer',
       'Middle management', 'Skilled labourers', 'Unskilled labourers',
       'Social class A', 'Social class B1', 'Social class B2',
       'Social class C', 'Social class D', 'Rented house', 'Home owners',
       '1 car', '2 cars', 'No car', 'National Health Service',
       'Private health insurance', 'Income 30.000', 'Income 30-45.000',
       'Income 45-75.000', 'Income 75-122.000', 'Income>123.000',
       'Average income', 'Purchasing power class',
       'Contribution private third party insurance',
       'Contribution third party insurance (firms)',

<p class=head>Type of Insurance</p>

In [36]:
fig=single_col_pct_chart(sociademographic, 'National Health Service')
fig.update_xaxes(categoryorder='array', categoryarray= [ '0%', '1 - 10%', '11 - 23%','24 - 36%','37 - 49%', '50 - 62%','63 - 75%','76 - 88%', '89 - 99%', '100%'])
fig.update_layout(height=500, width=700)
fig.show(renderer='colab')
fig=single_col_pct_chart(sociademographic, 'Private health insurance')
fig.update_xaxes(categoryorder='array', categoryarray= [ '0%', '1 - 10%', '11 - 23%','24 - 36%','37 - 49%', '50 - 62%','63 - 75%','76 - 88%', '89 - 99%', '100%'])
fig.update_layout(height=500, width=700)
fig.show(renderer='colab')
fig=col_pct_chart_w_target(sociademographic, 'National Health Service')
fig.update_xaxes(categoryorder='array', categoryarray= [ '0%', '1 - 10%', '11 - 23%','24 - 36%','37 - 49%', '50 - 62%','63 - 75%','76 - 88%', '89 - 99%', '100%'])
fig.update_layout(height=500, width=700)
fig.show(renderer='colab')
fig=col_pct_chart_w_target(sociademographic, 'Private health insurance')
fig.update_xaxes(categoryorder='array', categoryarray= [ '0%', '1 - 10%', '11 - 23%','24 - 36%','37 - 49%', '50 - 62%','63 - 75%','76 - 88%', '89 - 99%', '100%'])
fig.update_layout(height=500, width=700)
fig.show(renderer='colab')

<p class=gen>While customers used National Health Service present with higher percentage than those used Private Health Service, Private Health Service consumers have higher percentage of caravan insurance buyers.</p>

<p class=head>Income Levels and Average Income</p>

In [37]:
fig=single_col_pct_chart(sociademographic, 'Income 30.000')
fig.update_xaxes(categoryorder='array', categoryarray= [ '0%', '1 - 10%', '11 - 23%','24 - 36%','37 - 49%', '50 - 62%','63 - 75%','76 - 88%', '89 - 99%', '100%'])
fig.update_layout(height=500, width=700)
fig.show(renderer='colab')
fig=single_col_pct_chart(sociademographic, 'Income 30-45.000')
fig.update_xaxes(categoryorder='array', categoryarray= [ '0%', '1 - 10%', '11 - 23%','24 - 36%','37 - 49%', '50 - 62%','63 - 75%','76 - 88%', '89 - 99%', '100%'])
fig.update_layout(height=500, width=700)
fig.show(renderer='colab')
fig=single_col_pct_chart(sociademographic, 'Income 45-75.000')
fig.update_xaxes(categoryorder='array', categoryarray= [ '0%', '1 - 10%', '11 - 23%','24 - 36%','37 - 49%', '50 - 62%','63 - 75%','76 - 88%', '89 - 99%', '100%'])
fig.update_layout(height=500, width=700)
fig.show(renderer='colab')
fig=single_col_pct_chart(sociademographic, 'Income 75-122.000')
fig.update_xaxes(categoryorder='array', categoryarray= [ '0%', '1 - 10%', '11 - 23%','24 - 36%','37 - 49%', '50 - 62%','63 - 75%','76 - 88%', '89 - 99%', '100%'])
fig.update_layout(height=500, width=700)
fig.show(renderer='colab')

fig=col_pct_chart_w_target(sociademographic, 'Income 30.000')
fig.update_xaxes(categoryorder='array', categoryarray= [ '0%', '1 - 10%', '11 - 23%','24 - 36%','37 - 49%', '50 - 62%','63 - 75%','76 - 88%', '89 - 99%', '100%'])
fig.update_layout(height=500, width=700)
fig.show(renderer='colab')
fig=col_pct_chart_w_target(sociademographic, 'Income 30-45.000')
fig.update_xaxes(categoryorder='array', categoryarray= [ '0%', '1 - 10%', '11 - 23%','24 - 36%','37 - 49%', '50 - 62%','63 - 75%','76 - 88%', '89 - 99%', '100%'])
fig.update_layout(height=500, width=700)
fig.show(renderer='colab')
fig=col_pct_chart_w_target(sociademographic, 'Income 45-75.000')
fig.update_xaxes(categoryorder='array', categoryarray= [ '0%', '1 - 10%', '11 - 23%','24 - 36%','37 - 49%', '50 - 62%','63 - 75%','76 - 88%', '89 - 99%', '100%'])
fig.update_layout(height=500, width=700)
fig.show(renderer='colab')
fig=col_pct_chart_w_target(sociademographic, 'Income 75-122.000')
fig.update_xaxes(categoryorder='array', categoryarray= [ '0%', '1 - 10%', '11 - 23%','24 - 36%','37 - 49%', '50 - 62%','63 - 75%','76 - 88%', '89 - 99%', '100%'])
fig.update_layout(height=500, width=700)
fig.show(renderer='colab')

<p class=gen>Number of customers who income about 30.000-45.000 or 45.000-75.000 is higher than other income levels. The higher income customer gain, the higher ability to buy caravan insurance.</p>

In [38]:
fig=single_col_pct_chart(sociademographic, 'Purchasing power class')
fig.update_layout(height=500, width=700)
fig.update_xaxes(categoryorder='array', categoryarray= [  '1 - 10%', '11 - 23%','24 - 36%','37 - 49%', '50 - 62%','63 - 75%','76 - 88%', '89 - 99%', '100%'])
fig.show(renderer='colab')
fig=single_col_pct_chart(sociademographic, 'Average income')
fig.update_layout(height=500, width=700)
fig.show(renderer='colab')
fig=col_pct_chart_w_target(sociademographic, 'Purchasing power class')
fig.update_xaxes(categoryorder='array', categoryarray= [  '1 - 10%', '11 - 23%','24 - 36%','37 - 49%', '50 - 62%','63 - 75%','76 - 88%', '89 - 99%', '100%'])
fig.update_layout(height=550, width=700)
fig.show(renderer='colab')
fig=col_pct_chart_w_target(sociademographic, 'Average income')
fig.update_xaxes(categoryorder='array', categoryarray= [  '1 - 10%', '11 - 23%','24 - 36%','37 - 49%', '50 - 62%','63 - 75%','76 - 88%', '89 - 99%', '100%'])
fig.update_layout(height=500, width=700)
fig.show(renderer='colab')

<p class=gen> In term of Buyers' purchasing power, Most of consumers (>50% of 5822) has the ratio of purchasing power of 24-36% (26.28%), 37-49% and 63-75% (>15.4%). The higher level of purchasing power, the more percentage consumers bought caravan insurance policies.</p>
<p class=gen> The higher average income the higher rate of caravan buyers.</p>

<h3 class=head>4. Characteristic of Buyers Groups</h3>

In [39]:
buyers = train_dataset[train_dataset.Target=='insurance']

def pie_of_buyers(df, group_feature, col_for_count):
    fig=px.pie(data_frame=buyers.groupby(group_feature).count()[col_for_count].sort_values(ascending=False).reset_index(), values=col_for_count,labels=group_feature, names=group_feature, color_discrete_sequence=px.colors.qualitative.Bold, title=group_feature + ' among Buyers')
    fig.update_layout(font_family="Times New Roman",
                 font_size=14,
                 title_font_family="Times New Roman",
                 title_font_color='darkblue', height=500,width=500)
    return fig
# fig.write_image('images/target2.png')

In [40]:
fig = pie_of_buyers(buyers, 'Customer main type', 'Farmer' )
fig.show(renderer='colab')

In [41]:
fig = pie_of_buyers(buyers, 'Customer Subtype', 'Farmer' )
fig.update_layout(height=800, width=800)
fig.show(renderer='colab')